src/models/StyledComponent.js

```diff
@@ -64,7 +69,27 @@ export default (ComponentStyle: Function) => {
 
       generateAndInjectStyles(theme: any, props: any) {
         const executionContext = this.buildExecutionContext(theme, props)
-        return componentStyle.generateAndInjectStyles(executionContext)
+        const css = componentStyle.generateAndInjectStyles(executionContext)
+
+        const inlineRules = Object.keys(this.attrs).reduce((styles, name) => {
+          if (!Array.isArray(this.attrs[name])) {
+            return styles
+          }
+
+          if (!this.attrs[name].isStyledBlock) {
+            return styles
```

Comment:
@ramitos can you explain why this needs to be handled differently here? Shouldn't it just be able to come through as an array?